src/shared/modules.mjs

```diff
@@ -854,6 +854,167 @@ const generateEventListenResponse = json => {
+    }
+    return outType
+}
+
+const generateAnyOfSchema = (anyOf, name, summary) => {
+    let anyOfType = []
+    if (!anyOf['$ref']) {
+        anyOf.name = anyOf.title || anyOf.name || ''
+        delete anyOf.title
+        if (anyOf.schema === undefined) {
+            anyOf.schema = {}
+            if (anyOf.oneOf) {
+                anyOf.schema.oneOf = anyOf.oneOf
+                delete anyOf.oneOf
+            }
+            else if (anyOf.anyOf) {
+                anyOf.schema.anyOf = anyOf.anyOf
+                delete anyOf.anyOf
+            }
+            else if (anyOf.allOf) {
+                anyOf.schema.allOf = anyOf.allOf
+                delete anyOf.allOf
+            }
+        }
+    }
+    anyOfType.name = name[0].toLowerCase() + name.substr(1)
+    anyOfType.summary = summary
+    anyOfType.schema = anyOf
+    return anyOfType
+}
+
+const generateParamsAnyOfSchema = (methodParams, anyOf, anyOfTypes, title, summary) => {
+    let params = []
+    methodParams.forEach(p => {
+        if (p.schema === anyOfTypes) {
+            let anyOfType = generateAnyOfSchema(anyOf, title, summary)
+            anyOfType.required = p.required
+            params.push(anyOfType)
+        }
+        else {
+            params.push(p)
+        }
+    })
+    return params
+}
+
+const generateResultAnyOfSchema = (method, methodResult, anyOf, anyOfTypes, title, summary) => {
+    let methodResultSchema = Object.assign({}, method.result.schema)
+
+    if (methodResult.schema === anyOfTypes) {
+        let anyOfType = generateAnyOfSchema(anyOf, title, summary)
+        let index = 0
+        if (isEventMethod(method)) {
+            index = (method.result.schema.anyOf || method.result.schema.oneOf).indexOf(getPayloadFromEvent(method))
+        }
+        else {
+            index = (method.result.schema.anyOf || method.result.schema.oneOf).indexOf(anyOfType)
+        }
+        if (method.result.schema.anyOf) {
+            methodResultSchema.anyOf = Object.assign([], method.result.schema.anyOf)
+            methodResultSchema.anyOf[index] = anyOfType.schema
+        }
+        else if (method.result.schema.oneOf) {
+            methodResultSchema.oneyOf = Object.assign([], method.result.schema.oneOf)
+            methodResultSchema.oneOf[index] = anyOfType.schema
+        }
+        else {
+            methodResultSchema = anyOfType.schema
+        }
+    }
+    return methodResultSchema
+}
+
+const createPolymorphicMethods = (method, json) => {
+    let anyOfTypes = []
+    let methodParams = []
+    let methodResult = Object.assign({}, method.result)
+
+    method.params.forEach(p => {
+        if (p.schema) {
+            methodParams.push(getAnyOfSchema(p, json, anyOfTypes))
+        }
+    })
+    if (isEventMethod(method)) {
+        methodResult.schema = getPayloadFromEvent(method)
+    }
+    methodResult = getAnyOfSchema(methodResult, json, anyOfTypes)
+
+    let polymorphicMethodSchemas = []
+    if (anyOfTypes.length > 0) {
```

Comment:
Hmmm, I think we need separate logic for anyOf in the params vs the result.
Maybe you're handling it somehow, but it looks like you've got all anyOfs (from both params and results) in a single array, and don't know where they come from anymore...

Reply:
yea, but I think if there is an anyOf case in both params and result that is not valid case. so there is only possibility to have this in one place. hence why used single variable to handle it